salt/metalk8s/addons/monitoring/prometheus/deployed.sls

```diff
@@ -27,7 +27,7 @@ Expose Prometheus:
           port: 9090
           protocol: TCP
           node_port: 30222
-          targetPort: api
+          targetPort: web
```

Comment:
No. Upstream is definitely 'standard K8s YAML', i.e.`nodePort`. This `deployed.sls` is *not* from upstream, but added by us to create the `NodePort` *Service* for Prometheus. As such:
- `node_port` was used 'to make it work' (I guess)
- Due to the bug in the name handling, `targetPort` was simply ignored
- Things still 'worked' because the (numeric) `targetPort` is 9090, and `targetPort` defaults to `port` if not set (and even though it's set here, it wasn't passed to K8s API, so that one rightfully did what it's supposed to do)
- Due to my fix, `targetPort` is no longer not passed (heh), so it was set to the named port `api`, but our Prometheus *Pod*s don't *have* a port named `api`, so this was incorrect from the beginning.